contracts/KyberPoolMaster.sol

```diff
@@ -298,88 +403,211 @@ contract KyberPoolMaster is Ownable {
     }
 
     /**
-     * @dev  Claims rewards and distribute fees and its share to poolMaster
-     * @param epoch for which rewards are being claimed
+     * @dev  Queries the epochs with at least one feeHandler paying rewards, for the pool
      */
-    function claimRewardsMaster(uint256 epoch) public {
-        require(!claimedPoolReward[epoch], "cRMaster: rewards already claimed");
-
-        uint256 unclaimed = getUnclaimedRewards(epoch);
-
-        require(unclaimed > 0, "cRMaster: no rewards to claim");
-
-        uint256 initialBalance = address(this).balance;
-
-        kyberFeeHandler.claimStakerReward(address(this), epoch);
-
-        uint256 totalRewards = address(this).balance.sub(initialBalance);
-
-        (uint256 stake, uint256 delegatedStake, ) = kyberStaking
-            .getStakerRawData(address(this), epoch);
+    function getAllEpochWithUnclaimedRewards()
+        external
+        view
+        returns (uint256[] memory)
+    {
+        uint256 currentEpoch = kyberDao.getCurrentEpochNumber();
+        uint256 maxEpochNumber = currentEpoch.sub(firstEpoch);
+        uint256[] memory epochsWithRewards = new uint256[](maxEpochNumber);
+        uint256 epochCounter = 0;
+        for (uint256 epoch = firstEpoch; epoch <= currentEpoch; epoch++) {
+            for (uint256 i = 0; i < feeHandlersList.length; i++) {
+                uint256 unclaimed = getUnclaimedRewards(
+                    epoch,
+                    IExtendedKyberFeeHandler(feeHandlersList[i])
+                );
+
+                if (unclaimed > 0) {
+                    epochsWithRewards[epochCounter] = epoch;
+                    epochCounter++;
+                    // stop querying feeHandlers on the first with rewards for the epoch
+                    break;
+                }
+            }
+        }
 
-        DFeeData storage epochDFee = delegationFees[getEpochDFeeDataId(epoch)];
+        uint256[] memory result = new uint256[](epochCounter);
+        for (uint256 i = 0; i < epochCounter; i++) {
+            result[i] = epochsWithRewards[i];
+        }
 
-        uint256 totalFee = totalRewards.mul(epochDFee.fee).div(
-            MAX_DELEGATION_FEE
-        );
-        uint256 rewardsAfterFee = totalRewards.sub(totalFee);
+        return result;
+    }
 
-        uint256 poolMembersShare = calculateRewardsShare(
-            delegatedStake,
-            stake.add(delegatedStake),
-            rewardsAfterFee
-        );
-        uint256 poolMasterShare = totalRewards.sub(poolMembersShare); // fee + poolMaster stake share
+    struct RewardInfo {
```

Comment:
Let's move the definition of this struct to the beginning of the file with the rest of the other structs